src/lifecycle/props-init.js

```diff
@@ -77,23 +81,15 @@ function createNativePropertyDefinition (name, opts) {
 
     // Link up the attribute.
     if (this[$connected]) {
-      syncPropToAttr(this, opts, name, false);
+      syncPropToAttr(this, propDef, false);
     }
   };
 
   return prop;
 }
 
+// todo: This is only used from unit tests
```

Comment:
Can we drop this code and rewrite the tests? IIRC the tests have a single function that call out to this, so that'd be a single refactor point, probably.